sdk/core/azure-core/src/http/winhttp/win_http_transport.cpp

```diff
@@ -4,18 +4,572 @@
+  {
+    CleanupHandlesAndThrow(
+        "Error while querying response headers.", sessionHandle, connectionHandle, requestHandle);
+  }
+
+  HttpStatusCode httpStatusCode = static_cast<HttpStatusCode>(statusCode);
+
+  // Get the optional reason phrase.
+  std::string reasonPhrase;
+
+  if (WinHttpQueryHeaders(
+          requestHandle,
+          WINHTTP_QUERY_STATUS_TEXT,
+          WINHTTP_HEADER_NAME_BY_INDEX,
+          outputBuffer.data(),
+          &sizeOfHeaders,
+          WINHTTP_NO_HEADER_INDEX))
+  {
+    start = outputBuffer.data();
+    reasonPhrase = WideStringToString(std::wstring(start, start + sizeOfHeaders / sizeof(WCHAR)));
+  }
+
+  DWORD dwContentLength = 0;
+  dwSize = sizeof(dwContentLength);
+
+  int64_t contentLength = 0;
+
+  // Get the content length as a number.
+  if (httpStatusCode != HttpStatusCode::NoContent)
+  {
+    if (!WinHttpQueryHeaders(
+            requestHandle,
+            WINHTTP_QUERY_CONTENT_LENGTH | WINHTTP_QUERY_FLAG_NUMBER,
+            WINHTTP_HEADER_NAME_BY_INDEX,
+            &dwContentLength,
+            &dwSize,
+            WINHTTP_NO_HEADER_INDEX))
+    {
+      contentLength = -1;
+
+      // if (!WinHttpQueryHeaders(
+      //        requestHandle,
+      //        WINHTTP_QUERY_TRANSFER_ENCODING,
+      //        WINHTTP_HEADER_NAME_BY_INDEX,
+      //        outputBuffer.data(),
+      //        &sizeOfHeaders,
+      //        WINHTTP_NO_HEADER_INDEX))
+      //{
+
+      //  WinHttpQueryHeaders(
+      //      requestHandle,
+      //      WINHTTP_QUERY_CONTENT_ENCODING,
+      //      WINHTTP_HEADER_NAME_BY_INDEX,
+      //      outputBuffer.data(),
+      //      &sizeOfHeaders,
+      //      WINHTTP_NO_HEADER_INDEX);
+
+      //  WinHttpQueryHeaders(
+      //      requestHandle,
+      //      WINHTTP_QUERY_CONTENT_TRANSFER_ENCODING,
+      //      WINHTTP_HEADER_NAME_BY_INDEX,
+      //      outputBuffer.data(),
+      //      &sizeOfHeaders,
+      //      WINHTTP_NO_HEADER_INDEX);
+
+      //  CleanupHandlesAndThrow(
+      //      "Error while querying response headers.", sessionHandle, connectionHandle,
+      //      requestHandle);
+      //}
+
+      // start = outputBuffer.data();
+      // std::string transferEncoding
+      //    = WideStringToStringASCII(start, start + sizeOfHeaders / sizeof(WCHAR));
+
+      // if (transferEncoding != "chunked")
+      //{
+      //  CleanupHandlesAndThrow(
+      //      "Error while querying response headers.", sessionHandle, connectionHandle,
+      //      requestHandle);
+      //}
+    }
+    else
+    {
+      contentLength = static_cast<int64_t>(dwContentLength);
+    }
+  }
+
+  auto stream = std::make_unique<Details::WinHttpStream>(
+      sessionHandle, connectionHandle, requestHandle, contentLength);
+
+  // Allocate the instance of the response on the heap with a shared ptr so this memory gets
+  // delegated outside the transport and will be eventually released.
+  auto rawResponse
```

Comment:
this needs to be a `std::create_unique` and then on the return statement, you need to use std::move()

Reply:
Is this change still needed for functional correctness?